### PR TITLE
feat: provide intellisense for Vue SFC files

### PR DESCRIPTION
**Pull request by AumyF**
This pull request enables intellisense for GraphQL templates in `<script>` and `<script setup>` in `.vue` SFC files.

# Example

```vue
<script setup lang="ts">
// graphql-codegen uses the query to generate some code
`#graphql
query {id}
`;
</script>
```

### CODE REVIEW

**Comment by acao** [comment]
so exciting @AumyF !! thanks for doing this!

@stonexer would you like to help review this as you have time?

File: packages/graphql-language-service-server/src/findGraphQLTags.ts
Comment (stonexer) [inline packages/graphql-language-service-server/src/findGraphQLTags.ts:119]: The line: 110 already has a `result`, another name here would be better.

**Review by stonexer — CHANGES_REQUESTED** [review]
Nice job! 

Maybe we can add some relevant tests?

**Comment by acao** [comment]
@AumyF it still wants you to sign the CLA, are you able to?

File: packages/graphql-language-service-server/src/findGraphQLTags.ts
Comment (acao) [inline packages/graphql-language-service-server/src/findGraphQLTags.ts:77]: perhaps it might be easier and more readable to declare this return type union above rather than inline?

**Comment by acao** [comment]
@AumyF anything I can do to help you finish this PR?

**Comment by acao** [comment]
@AumyF the issue with EasyCLA is easily fixable. You just need to sign the CLA using the email that *all* of your commits are signed with. Sorry for the boilerplate, your next PR will be much easier haha!

**Comment by acao** [comment]
@AumyF also can you add some tests? it might require a rebase, but we have some very specific tests for `findGraphQLTest` that are very easy to write and run

**Comment by acao** [comment]
@AumyF I am going to revive this as a new PR. If you're able to sign the CLA with the same email the commits are signed with this weekend I can give you actual commit credit, but if not, I can give you a git username link attribution in the changelog. Unfortunately our CLA rules are quite strict and I might get in trouble for merging the work of someone who didn't pass the annoying but necessary CLA bot

**Comment by AumyF** [comment]
I'm sorry for the late reply.  I have signed the CLA but I did with a different email address... The signature is invalid, isn't it?

And I finally got my motivation back, so, may I resume the pull request?

**Comment by acao** [comment]
@AumyF it would be an honor! it's a fantastic one and I'd be happy to help you see it through

So, you can either re-sign the CLA using the email address you _did_ use for every commit.

If you prefer not to have to sign the form again, you can perform a `git rebase -i` however you prefer in such a way as to rewrite the email addresses used in the commits to the one you originally signed the CLA with.

**Comment by AumyF** [comment]
I rebased onto latest main and added some commits (I will squash them later)

**Comment by acao** [comment]
@AumyF thank you sooo much! this is so exciting. looks like the very picky CLA bot is  finally happy!! 😆

File: packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts
Comment (acao) [inline packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts:193]: thank you for adding these tests!!

reading more about vue, i wonder if there are cases where .vue files are plain typescript/javascript and whether we should test for those - i `think` i see where you account for this in `findGraphQLTags`? If i understand correctly, render functions may be a case of this? or do these files have standard js/ts files in vue? https://vuejs.org/guide/extras/render-function.html#creating-vnodes

File: packages/vscode-graphql/esbuild.js
Comment (acao) [inline packages/vscode-graphql/esbuild.js:17]: ah, is there any way we can more selectively import it by chance?

this only removes it for `vscode-graphql` users, and we have vim, emacs, sublime users as well, who may end up with a bloated bundle over this. are they dynamically loaded into the runtime as needed at least?

File: packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts
Comment (AumyF) [reply packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts:193]: As far as I know there is no such cases. Though we can write Vue components as non-SFC files using render functions, they should have a .js, .jsx, .ts or .tsx extension.

However thanks to your comment I realized I forgot to handle JSX in SFC.  `@vue/compiler-sfc` seems to supports JSX out of the box, so I added a test for it.

File: packages/vscode-graphql/esbuild.js
Comment (AumyF) [reply packages/vscode-graphql/esbuild.js:17]: My comment was a bit misleading. I specified these packages as `external` not to reduce bundle size, but to avoid bundling errors. They are template preprocessors and `require()`d at runtime by `@vue/consolidate`. `@vue/consolidate` does not specify the preprocessors as `dependencies` in its package.json, so users need to install them separately.

However, in this case, esbuild tries to load the preprocessors and fails with `package not found` error. 

If the users just install and use `graphql-language-service-cli`, no extra packages installed.

File: packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts
Comment (acao) [reply packages/graphql-language-service-server/src/__tests__/findGraphQLTags-test.ts:193]: i think that's what i meant, rad thanks!

**Review by acao — APPROVED** [review]
This looks good to go to me!

**Comment by acao** [comment]
@AumyF I tried pushing a rebase but it says it's rejected, I wonder what I'm doing wrong? hmm!

**Comment by acao** [comment]
Screenshot 2022-08-24 at 21 40 20
even without --force! hmm!

**Comment by acao** [comment]
@tobiasdiez it is only missing a rebase! I was hoping to merge this after workspaces support, but I may merge it first because we had to revert workspaces

**Comment by acao** [comment]
@tobiasdiez #2827 will be the one - I'm not able to push the rebase to the PR, so I've re-created it.

**Pull request closed without merging**
